Controllers/CHumidityController.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * CHumidityController.cpp
+ *
+ *  Created on: Aug. 11, 2022
+ *      Author: samk
+ */
+
+#include "CHumidityController.h"
+#include <stdio.h>
+#include "CBME280.h"
+#include "CHumidifier.h"
+#include "ICommand.h"
+#include "IHardwareMap.h"
+
+#define MIN_TEMPERATURE  10
```

Comment:
Agreed. @samkno1, there are two ways you can set constants in your code:
`#define CONSTANT_VALUE 10`
or
`constexpr unsigned int CONSTANT_VALUE = 10;`

Fundamentally, they both achieve the same thing, as the values will be resolved at compile time, but `#define` is resolved by pre-processor that has no logic in it. It blindly replaces every instance of `CONSTANT_VALUE` with a number 10. While if you declare it as `const` or `constexpr` then compiler will make sure that types are correct. 
Generally, using `const` or `constexpr` is considered safer and a better practice. As far as memory usage goes, compiler will optimise the values out, so memory usage will not be impacted.
Finally, `const` vs. `constexpr` is a bit subtle. You can use `constexpr` everywhere. The difference was introduced in 2014 edition, IIRC, and allows you to not only designate values, but also expressions. Example:
`constexpr float HALF_ANGLE_SIN = sin(45);`
as long as the expression can be evaluated at compile time and not at runtime, it can be declared as `constexpr`.